hw/market.py

```diff
@@ -1,28 +1,50 @@
+from log_decorators import start_end_log_decorator
+from utils import convert_string_to_date
+
 class Market:
+    @start_end_log_decorator
     def __init__(self, wines: list = None, beers: list = None) -> None:
-        pass
-
+        self.wines = {wine.title: wine for wine in wines} if wines else {}
+        self.beers = {beer.title: beer for beer in beers} if beers else {}
+        self.allDrinks = (wines or []) + (beers or [])
+ 
+    @start_end_log_decorator
     def has_drink_with_title(self, title=None) -> bool:
         """
         Проверяет наличие напитка в магазине за О(1)
 
         :param title:
         :return: True|False
         """
-        pass
+        return title in self.wines or title in self.beers if title else bool
```

Comment:
Блин :)
Поправил
[fix: function should return False instead of bool class :)](https://github.com/ark-web-dev/hh-school-2023-python-core/pull/1/commits/d1c07005fc5681e67610a02f9b1f2c00c2d6ecf2)